specs/IFrameMemoryUsage.md

```diff
@@ -0,0 +1,306 @@
+  /// Use ICoreWebView2Frame.add_Destroyed to listen for when this iframe goes
+  /// away.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT add_FrameCreated(
+      [in] ICoreWebView2NestedFrameCreatedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with add_FrameCreated.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT remove_FrameCreated([in] EventRegistrationToken token);
+
+  /// Get the renderer process id of the iframe.
+  /// The renderer process ID can change when navigating to a new document 
+  /// and you can use the RendererProcessIdChanged event to know when it changes. 
+  /// Child frames may have different renderer processes. 
+  // MSOWNERS: wangsongjin@microsoft.com
+  [propget] HRESULT RendererProcessId([out, retval] UINT32* value);
+
+  /// Subscribe to renderer process Id changed event.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT add_RendererProcessIdChanged(
+      [in] ICoreWebView2FrameRendererProcessIdChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with add_RendererProcessIdChanged.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT remove_RendererProcessIdChanged([in] EventRegistrationToken token);
+}
+
+/// A continuation of the `ICoreWebView2` interface to support get 
+/// renderer process id and handle render process change event
+[uuid(ad712504-a66d-11ed-afa1-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2_18 : ICoreWebView2_17 {
+  /// Get the renderer process id of the main frame
+  /// The renderer process ID can change when navigating to a new document.
+  //  
+  // MSOWNERS: wangsongjin@microsoft.com
+    [propget] HRESULT RendererProcessId([out, retval] UINT32* value);
+
+  /// Subscribe to renderer process id changed event
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT add_RendererProcessIdChanged(
+      [in] ICoreWebView2RendererProcessIdChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with add_RendererProcessIdChanged.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT remove_RendererProcessIdChanged([in] EventRegistrationToken token);
+}
+```
+
+C#
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{   
+    runtimeclass CoreWebView2
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2_18")]
+        {
+            // ICoreWebView2_18 members
+            UInt32 RendererProcessId { get; };
+            event Windows.Foundation.TypedEventHandler<CoreWebView2, Object> RendererProcessIdChanged;
+        }
+    }
+
+    runtimeclass CoreWebView2Frame
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2Frame5")]
+        {
+            // ICoreWebView2Frame5 members
+            UInt32 RendererProcessId { get; };

```

Comment:
This and above ProcessId property should be Int32 not UInt32, but we're moving to ProcessInfo as described above so shouldn't matter.